### PR TITLE
Modularize scale-test

### PR DESCRIPTION
**Pull request by marseel**
Make scale-test more modular to allow for adding more scale-test scenarios.
Also, increase the frequency of node-throughput test as it only creates two VMs.

### CODE REVIEW

**Comment by marseel** [comment]
Successful run: https://github.com/cilium/cilium/actions/runs/6627684165/job/18003120946

File: .github/actions/scale-tests/create-cluster/action.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  kops_state:
+    description: "Bucket where to store kops state"
+    required: true
+  project_id:
+    description: "GCP Project in which to create cluster"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Check if we should build the Cilium CLI from source
```
Comment (learnitall) [inline .github/actions/scale-tests/create-cluster/action.yaml:28]: I think the name for this step is incorrect?

File: .github/actions/scale-tests/export-results/action.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  other_files:
+    description: "Additional files to export"
+    required: true
+  test_name:
+    description: "Name of the test"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Delete cluster
```
Comment (learnitall) [inline .github/actions/scale-tests/export-results/action.yaml:19]: I think the name for this step is incorrect here too.

File: .github/actions/scale-tests/install-kops/action.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: 'Install kops'
+description: "Workflow that installs kops cli"
+inputs:
+  release_version:
+    description: "version of kops"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Delete cluster
```
Comment (learnitall) [inline .github/actions/scale-tests/install-kops/action.yaml:10]: This step name seems off also.

File: .github/actions/scale-tests/validate-cluster/action.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: 'Validate cluster'
+description: "Workflow that wait for cluster to be in a ready state"
```
Comment (learnitall) [inline .github/actions/scale-tests/validate-cluster/action.yaml:2]:
```suggestion
description: "Workflow that waits for cluster to be in a ready state"
```

File: .github/workflows/scale-test-gce.yaml [hunk 5]
```diff
@@ -36,8 +36,8 @@ env:
   go_version: 1.21.3
   # Adding k8s.local to the end makes kops happy-
   # has stricter DNS naming requirements.
-  cluster_name: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}-${{ github.run_attempt }}.k8s.local
-  cl2_config: ${{ github.workspace }}/.github/actions/scale-test-gce/cl2-config.yaml
+  test_name: node-throughput
+  cluster_name: ${{ github.run_id }}-${{ github.run_attempt }}.k8s.local
```
Comment (learnitall) [inline .github/workflows/scale-test-gce.yaml:40]: Can you switch these two lines? I think the `cluster_name` should be directly underneath the comment about k8s.local.

File: .github/actions/scale-tests/validate-cluster/action.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Validate cluster
+      shell: bash
+      run: |
+        ./kops validate cluster \
+          ${{ inputs.cluster_name }} \
+          --state ${{ inputs.kops_state }} \
+          --wait 20m \
```
Comment (learnitall) [inline .github/actions/scale-tests/validate-cluster/action.yaml:19]: Nit: this timeout value is hard coded, but a user could specify a longer timeout when consuming the workflow through the GHA option `timeout-minutes`.

File: .github/workflows/scale-test-gce.yaml [hunk 7]
```diff
@@ -1,8 +1,8 @@
-name: Scale Test GCE
+name: Node throughput test
```
Comment (learnitall) [inline .github/workflows/scale-test-gce.yaml:1]: Maybe it would make sense to also rename the file?

**Review by learnitall — CHANGES_REQUESTED** [review]
After looking through this, I think it might be a good idea if we just make one large action that can be consumed, rather than splitting out each step into its own action. For instance, if we were to add another workflow that runs a different scale test, we would essentially copy and paste the node-throughput test and just edit the CL2 config and number of nodes. I'm not sure what the use case would be for consuming the smaller actions on their own. WDYT?

Besides that, I have a couple of small nits.

**Pull request closed without merging**